workflow/controller/operator.go

```diff
@@ -1822,6 +1822,18 @@ func (woc *wfOperationCtx) possiblyGetRetryChildNode(node *wfv1.NodeStatus) *wfv
 			if childNode == nil {
 				continue
 			}
+
+			// childNode.nodeFlag was missing, this can only happen if NodeStatus is old
+			if childNode.NodeFlag == nil {
+				// fallback where nodeFlags is absent
+				// ensure that we do not return
+				if !common.CheckHookNode(childNode.Name) {
+					return childNode
+				} else {
+					continue
```

Comment:
breaks 3.5.x code now, since any version before this PR might have NodeFlag be nil